presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/Util.java

```diff
@@ -105,6 +107,26 @@ public static PlanNode transpose(PlanNode parent, PlanNode child)
                         identityAssignmentsAsSymbolReferences(restrictedOutputs)));
     }
 
+    /**
+     * @return If the node has outputs not in permittedOutputs, returns an identity projection containing only those node outputs also in permittedOutputs in RowExpression.
+     */
+    public static Optional<PlanNode> addRowExpressionIdentityProjection(PlanNodeIdAllocator idAllocator, PlanNode node, Set<VariableReferenceExpression> permittedOutputs)
```

Comment:
Why don't we call it `restrictOutputs`?

Reply:
It will conflict with existing one since the signature is same.

Reply:
Put them with similar names to indicate the types of output. The two functions are doing the same thing; they should have similar names.